test/e2e/framework/framework.go

```diff
@@ -0,0 +1,36 @@
+package framework
+
+import (
+	"flag"
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"k8s.io/client-go/rest"
+)
+
+var Global *Framework
+
+type Framework struct {
+	KubeConfig *rest.Config
+	KubeClient kubernetes.Interface
+}
+
+func setup() error {
+	config := flag.String("kubeconfig", os.Getenv("HOME")+"/.kube/config", "kube config path, e.g. $HOME/.kube/config")
```

Comment:
What if `os.Getenv("HOME")` doesn't exist?

Prefer to check env var first `os.LookupEnv("HOME")` before using it in `flag.String(...)`.